packages/medusa/src/api/routes/store/products/list-products.ts

```diff
@@ -201,15 +199,12 @@ export default async (req, res) => {
   // get only published products for store endpoint
   filterableFields["status"] = ["published"]
 
-  const featureFlagRouter: FlagRouter = req.scope.resolve("featureFlagRouter")
-  if (featureFlagRouter.isFeatureEnabled(PublishableAPIKeysFeatureFlag.key)) {
-    if (req.publishableApiKeyScopes?.sales_channel_id.length) {
-      filterableFields.sales_channel_id =
-        filterableFields.sales_channel_id ||
-        req.publishableApiKeyScopes.sales_channel_id
+  if (req.publishableApiKeyScopes?.sales_channel_id.length) {
+    filterableFields.sales_channel_id =
+      filterableFields.sales_channel_id ||
+      req.publishableApiKeyScopes.sales_channel_id
 
-      listConfig.relations.push("sales_channels")
-    }
+    listConfig.relations.push("sales_channels")
```

Comment:
todo: here we push the relation only if `req.publishableApiKeyScopes?.sales_channel_id.length` but if it is provided by the user and there is not publishable then the relation is not pushed. Should we extract that and check if one or the other push the relation?

Reply:
this is a store products endpoint which doesn't have `sales_channels` in default/allowed relations ATM so user couldn't pass `sales_channels` for now, should we change that or keep it as is @olivermrbl?

but even in case that the user could provided `expand=sales_channels`, wouldn't we have this in `listConfig.relations` already? So we actually need to check if we are not pushing this relation twice.

Reply:
yes indeed, we should also check that the relation is not present as well 👍

Reply:
Think it makes sense to add sales_channels to allowed relations for products 👍